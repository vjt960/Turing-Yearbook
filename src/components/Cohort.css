h2 {
  background-color: rgba(255, 255, 255, 0.95);
  padding: 10px;
  color: hsl(218, 78%, 67%);
}

.cohort-grid {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 10px;
  margin: 1%;
}