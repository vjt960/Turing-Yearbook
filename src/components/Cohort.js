import React from 'react';
import Person from './Person';
import './Cohort.css';
import App from './App';

// COHORT COMPONENT CODE GOES HERE
const Cohort = ({ title, people }) => {
  const group = people.map(person => {
    return (
      <Person 
      image={person.photo} 
      name={person.name} 
      quote={person.quote} 
      superlative={person.superlative} 
      key={people.indexOf(person)}
      />
    )
  });
  return (
    <section className='cohort-block'>
      <h2>{title}</h2>
      <div className='cohort-grid'>
        {group}
      </div>
    </section>
    );
};

export default Cohort;
