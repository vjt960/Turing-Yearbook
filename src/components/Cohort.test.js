import React from 'react';
import { shallow, mount } from 'enzyme';
import Cohort from './Cohort';

describe('Cohort', () => {

  let mockData;
  beforeEach(() => {
    mockData = [
      {id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'},
      {id: 2, name: 'Amy', quote: 'I believe in you, cuties!', superlative: 'Most Likely to Be an Actual Superhero', photo: './amy.jpg'},
      {id: 3, name: 'Khalid', quote: 'Don\'t put the world on your shoulders.', superlative: 'Most Likely to Save the Planet from Alien Invasion', photo: './khalid.png'},
      {id: 4, name: 'Travis', quote: 'You do you.', superlative: 'Most Likely to Win a Grammy', photo: './travis.jpg'},
      {id: 5, name: 'Leta', quote: 'It\'s okay to be kind of a mess.', superlative: 'Most Likely to Go to the Moon', photo: './leta.jpg'}
    ];
  })

  it('has the right props', () => {
    const wrapper = mount(<Cohort  
      title='Staff' 
      people={mockData} 
      />);

      const mockObject = {id: 1, name: 'Eric', quote: 'You miss all the shots you don\'t take.', superlative: 'Most Likely to Win the PGA', photo: './eric.jpg'};

      expect(wrapper.prop('people')).toContainEqual(mockObject);
  });

  it('renders all props as Person components', () => {
    const wrapper = mount(<Cohort  
      title='Staff' 
      people={mockData} 
      />);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Cohort  
      title='Staff' 
      people={mockData} 
      />);
      expect(wrapper).toMatchSnapshot();
  })
})