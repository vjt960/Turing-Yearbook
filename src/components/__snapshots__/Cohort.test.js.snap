// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot 1`] = `ShallowWrapper {}`;
